apps/app_rpt/rpt_bridging.c

```diff
@@ -503,12 +500,13 @@ int rpt_call_bridge_setup(struct rpt *myrpt, struct ast_channel *mychannel, stru
 
 	/* put vox channel monitoring on the channel  
 	 *
-	 * The conference flags were originally DAHDI_CONF_MONITOR.  This 
-	 * setting caused dahdi_conf_add to return EINVAL.  The  flags
-	 * RPT_CONF_CONF | RPT_CONF_LISTENER permits the vox channel to 
-	 * join the conference.
+	 * This uses the internal DAHDI channel number to create the 
+	 * monitor conference.  It requires the ASL version of DAHDI
+	 * that contains changes for this to work properly.  
```

Comment:
Is there any way for us to check that we're using a compatible version? For example, can we update the DAHDI version string and have [app_rpt] request/check the string (w/DAHDI_GETVERSION)?

Note : it looks like our "dkms" version adds "+asl3" to the string (e.g. "3.3.0+asl3") ... Maybe that's something we could check?

For more info, see AllStarLink/dahdi-linux/debian/patches/version.patch

Reply:
There is an ioctl to get the dahdi version along with other information.  I had thought about this.  That would allow us to jump over the code that causes the hang.  

I did not know how deal with changing the version number.  We might need to consult with @jxmx and get his thoughts.

If @InterLinked1 gets this merged up stream, we will not have a problem.  He was planning on adding it to the phreaknet script.

Reply:
> There is an ioctl to get the dahdi version along with other information. I had thought about this. That would allow us to jump over the code that causes the hang.
> 
> I did not know how deal with changing the version number. We might need to consult with @jxmx and get his thoughts.
> 
> If @InterLinked1 gets this merged up stream, we will not have a problem. He was planning on adding it to the phreaknet script.

Hold on a second... I don't want to see the version number changed here (at least for source), the version number is set upstream and changing that would just lead to confusion.

Now that the patch is confirmed working, I'll get it in PhreakScript this week so that future installs from source won't need to worry about anything. Only existing users would be impacted (I assume the packages would also be updated at the same time) and a simple upgrade will fix it for them. Accordingly, the text "It requires the ASL version of DAHDI that contains changes for this to work properly." can be removed, because we're not maintaining a fork, it's just a bugfix that needs to make its way upstream, but until then both source install and packages will have the fix.

If Allan was talking about packaging versioning, I don't see any issue with that, I'm just talking about in the code itself.

Reply:
> If Allan was talking about packaging versioning, I don't see any issue with that, I'm just talking about in the code itself.

The current version of Asterisk/dahdi-linux is 3.4.0.  This version of the code does not have the fixes that we need.  If you add a patch to PhreakScript would the version change?  If not, then how would we know if the version of DAHDI installed on the system includes the fix(es) that we need?  The bottom line is that we don't want asterisk to crash nor hang.

AllStarLink/dahdi-linux currently updates the version string from "3.3.0" to "3.3.0+asl3".  That extension is something we can query/check.  In a similar fashion, PhreakScript could also tweak the version string to indicate that it has also applied patches. My proposal is that we update app_rpt to check the DAHDI version string.  If the version is one that we know includes the needed patches in the base (not with patches) then we're good (e.g. >3.5.0).  Else, we should check that the version string has a suffix that we're confident that it has the needed patches (e.g. "+PhreakScript" and "+asl3" are good).

Reply:
What is it that would actually be querying/checking? If it's a compile time check, there is probably a way we could check it in the build system. If it's app_rpt itself, maybe a macro define or something like that could be used?

PhreakScript does not change the version, because apart from a few minor tweaks to Asterisk/DAHDI, it is mostly adding functionality, and it just uses the base version. Some people call it "PhreakNet Asterisk" to differentiate from "vanilla" Asterisk but there isn't a formal difference really.

Reply:
I was thinking a runtime check (using ioctl(..., DAHDI_GETVERSION, ...).  That way we'd know that we're interacting with a version of DAHDI with the needed changes.  You could use the same type of check to know that the installed version of DAHDI with the "PhreakNet Asterisk" extras.

But, thinking about this more, this may not be a real issue.  If folks are using the ASL3 packages then we can ensure they are in sync.  If folks are building both DAHDI and Asterisk with PhreakScript they'll also be OK.  The exception would be for anyone building Asterisk+app_rpt that does not also build/install DAHDI ... and those who start with the ASL3 DAHDI, use dpkg to remove the .deb, and then install an incompatible version.

Reply:
> I was thinking a runtime check (using ioctl(..., DAHDI_GETVERSION, ...). That way we'd know that we're interacting with a version of DAHDI with the needed changes. You could use the same type of check to know that the installed version of DAHDI with the "PhreakNet Asterisk" extras.
> 
> But, thinking about this more, this may not be a real issue. If folks are using the ASL3 packages then we can ensure they are in sync. If folks are building both DAHDI and Asterisk with PhreakScript they'll also be OK. The exception would be for anyone building Asterisk+app_rpt that does not also build/install DAHDI ... and those who start with the ASL3 DAHDI, use dpkg to remove the .deb, and then install an incompatible version.

Yup, that was sort of my thinking too... anyone that installs with either method should be kept in sync.